lib/bindevict.go

```diff
@@ -77,37 +78,39 @@ func NormalizeBindName(bindName0 string) string {
 
 func (entry *BindThrottle) decrAllowEveryX(y int) {
 	if y >= 2 && logger.GetLogger().V(logger.Warning) {
-		info := fmt.Sprintf("hash:%d bindName:%s val:%s allowEveryX:%d-%d",entry.Sqlhash, entry.Name, entry.Value, entry.AllowEveryX, y)
+		info := fmt.Sprintf("hash:%d bindName:%s val:%s allowEveryX:%d-%d", entry.Sqlhash, entry.Name, entry.Value, entry.AllowEveryX, y)
 		logger.GetLogger().Log(logger.Warning, "bind throttle decr", info)
 	}
 	entry.AllowEveryX -= y
 	if entry.AllowEveryX > 0 {
 		return
 	}
 	entry.AllowEveryX = 0
-	GetBindEvict().lock.Lock()
-	defer GetBindEvict().lock.Unlock()
+	bindEvict := GetBindEvict()
+	bindEvict.lock.Lock()
+	defer bindEvict.lock.Unlock()
 	// delete entry
-	if len(GetBindEvict().BindThrottle[entry.Sqlhash]) == 1 {
-		updateCopy := GetBindEvict().Copy()
+	if len(bindEvict.BindThrottle[entry.Sqlhash]) == 1 {
+		updateCopy := bindEvict.Copy()
 		delete(updateCopy.BindThrottle, entry.Sqlhash)
 		gBindEvict.Store(updateCopy)
 	} else {
 		// copy everything except bindKV (skipping it is deleting it)
 		bindKV := fmt.Sprintf("%s|%s", entry.Name, entry.Value)
 		updateCopy := make(map[string]*BindThrottle)
-		for k,v := range GetBindEvict().BindThrottle[entry.Sqlhash] {
+		for k, v := range bindEvict.BindThrottle[entry.Sqlhash] {
 			if k == bindKV {
 				continue
 			}
 			updateCopy[k] = v
 		}
-		GetBindEvict().BindThrottle[entry.Sqlhash] = updateCopy
+		gBindEvict.Store(bindEvict)
```

Comment:
You have to store the updateCopy in the bindEvict.BindThrottle. 

gBindEvict.Store(bindEvict) is essentially storing everything and doesn't delete the entry.

Reply:
updated the code